### ISSUE TITLE
Adding a new page doesn't update `ctx.getFiles` calls in the watcher

### ISSUE THREAD

**Issue by Sciman101** [Osmose/phantomake#2]
**Problem**
Let's say I have a site with a directory structure like this.
```
blog/
  index.ejs
  post-1/
    index.md
  post-2/
    index.md
  (etc)
```
The blog's `index.ejs` fetches a list of blog posts using `ctx.getFiles` to display, and watches those files so if they change, it gets reloaded by the watcher. However, if I added a new post with `index.md` to the `blog` directory, it isn't caught in the same process and the watcher doesn't update.

**Ideal Solution**
It would be nice if the watcher could somehow look for pages being added that satisfied a getFiles call, and add them to the dependency list while running. Not sure how complex this would be in practice.

**Comment by Osmose** [comment]
Yeah, the watcher should be able to store all the registered globs and check if any newly-introduced file matches them, and hook it into the dependency tree. Thanks for the repoooooorrrrttttttt

**Comment by Osmose** [comment]
I spent a few hours hacking away on the smart version of this (i.e. tracking the globs and rechecking them as needed) but it was starting to get difficult to follow and annoying to deal with.

And I realized I could also just _always_ rebuild any file with a glob in it. This has the potential to cause slowness in the watch for pages that use globs that check a _ton_ of files, or for sites with lots of pages with globs in them, but.... eh. When someone actually hits that limitation we can deal with it then. For now it functions correctly, at least.

@Sciman101 This should be available in the [new 0.6.0 version](https://github.com/Osmose/phantomake/releases/tag/v0.6.0), lemme know if this fixes your issue!
